internal/compiler/wasm/wasm.go

```diff
@@ -1006,15 +1012,14 @@ func (c *Compiler) compileUpsert(local ir.Local, path []int, value ir.Local, ins
 
 		// If the next node is not an object, generate a conflict error.
 		inner = append(inner, instruction.Block{
-			Instrs: []instruction.Instruction{
+			Instrs: append([]instruction.Instruction{
 				instruction.GetLocal{Index: ltemp},
 				instruction.Call{Index: c.function(opaValueType)},
 				instruction.I32Const{Value: opaTypeObject},
 				instruction.I32Eq{},
 				instruction.BrIf{Index: 0},
-				instruction.I32Const{Value: c.builtinStringAddr(errWithConflict)},
-				instruction.Call{Index: c.function(opaAbort)},
 			},
+			c.runtimeErrorAbort(loc, errWithConflict)...),
```

Comment:
⚠️ This one didn't include `Unreachable` before, and now does. However, I suppose this code path will be revisited in #2922, too.